fili-core/src/main/java/com/yahoo/bard/webservice/druid/model/filter/SearchFilter.java

```diff
@@ -42,6 +42,22 @@
         QueryType(String type) {
             this.type = type;
         }
+
+        /**
+         * Get the QueryType enum from it's search type.
+         *
+         * @param type  The type of query (i.e. "insensitive_contains")
+         *
+         * @return the enum QueryType
+         */
+        public static QueryType fromType(String type) {
```

Comment:
I'm not sure I understand what you mean. Do you want me to enumerate each possibility to find which type the enum is instead of using the loop?

From the second example you gave they did this which is nearly identical to the way I checked the type.

```java
public static Status fromStatusCode(final int statusCode) {
    for (Status s : Status.values()) {
        if (s.code == statusCode) {
            return s;
        }
    }
    return null;
}
```